utils/docker/Dockerfile.centos.8

```diff
@@ -43,9 +45,13 @@ gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
           -i /etc/dnf/dnf.conf;                                 \
       dnf -y upgrade epel-release;                              \
     else                                                        \
-      dnf -y install epel-release dnf-plugins-core &&           \
+      dnf -y install dnf-plugins-core &&                        \
       dnf config-manager --assumeyes --set-enabled powertools;  \
+      dnf -y install epel-release;                              \
     fi;                                                         \
+    dnf config-manager --save --setopt=assumeyes=True &&        \
+    dnf config-manager --save --setopt=best=False &&            \
```

Comment:
It's probably a problem with everything other than CentOS, as epel isn't in sync with anything else.  This does solve (or at least appear to) the occasional deps issues in rocky however by sometimes installing a slightly older package version.

Part of the change in this PR is that if things are working well only master will create the images, so the package install and deps build will be skipped entirely for PRs, and PRs will use the images from the most successful master build, so if this does fail then it'll fail on master and PRs will simply continue to pull images from the cache from the previous master build.

Reply:
Rocky shouldn't fail any more than CentOS or Alma do.  They are all supposed to be binary compatible with each other because they are all supposed to be rebuilds and binary compatible with RHEL.  If there is a genuine sync problem with EPEL and EL8, then they all should have the same failures.

I think what this is telling us is that Rocky isn't maintaining up-to-date binary compatibility with EL and is a poor choice (at this point in time at least)  for a CentOS Linux replacement.

Moreover, I don't think we should be papering over this issue.  It is vitally important to us to understand which EL downstreams are stable and which are not.  The future of our own ability to produce stable packages for EL may depend on knowing this.

Reply:
they might be binary compatible with each other, but that doesn't mean the packages are in sync and there appear to be times when epel has been updated but Rocky hasn't.  I suspect the answer is to not use epel with Rocky but until/unless they have an alternative source of the packages we're using (and they may already - I haven't looked) then I think we're going to continue to get sporadic failures.

This is somewhat tangential to the rest of the PR, this change was an attempt to add stability in that it now works in cases where it would sometimes fail, but the rest of this PR is working so well - and it's only allowing PR builds on known-good docker images (assuming they don't get cache evicted) that actually the problem would only affect master build and not PR builds once this is landed, so we could easily remove this change while we continue to investigate a centos replacement.

Reply:
Ultimately, Rocky broke and CentOS and Alma didn't.  Clearly Rocky are doing something wrong.  They are the outlier.

But more importantly, I don't want to paper over these failures.  They are vitally important in our assessment of which distro to build on come Feb. 1.

Reply:
I've also seen CentOS fail but you're correct, I'll remove this part of the PR.

Reply:
https://github.com/ashleypittman/daos/runs/4224837764?check_suite_focus=true is the link for CentOS 8 failing although it's not quite what I thought it was, as best=False was in use and it still didn't resolve the issue due to 404 errors downloading the update.  Interestingly though it had installed everything, then when it went to update it the update itself failed even though it should have been basically a noop unless there was any packages updated in the previous ~5 minutes.

I've backed out this part of the change now, and as I say part of this PR is that PR builds will no longer depend on upstream repos if the github cache is working properly.